catalog/ui/search-ui/standard/src/main/webapp/js/view/preferences/PreferencesModal.view.js

```diff
@@ -99,8 +99,55 @@ define([
                 this.modelBinder.bind(this.viewModel, this.$el, this.colorBindings);
                 this.colorButtonsRegion.show(this.colorButtons);
             },
+            initSpectrum: function () {
+                /*
+                 the following fixes color input types for IE/chrome/firefox.
+
+                 - see http://caniuse.com/#feat=input-color
+
+                 - spectrum feature detects (chrome/firefox) native support for input[type=color],
+                 and spectrum will NOT "replace" these inputs.
+
+                 - IE has no native support for input[type=color], and the require of spectrum
+                 causes these inputs to be hidden, and spectrum builds "replacer" divs
+                 and the color pickers.
+
+                 - spectrum is implemented as a auto-executed anonymous function. This, plus
+                 semantics of require, means that spectrum only runs once per user session.
+                 Thus (w/o this fix), when a user opens the pref modal more than once, the
+                 input elements show only the color hex value as text.
+
+                 - also fixes native input color picker bug (chrome/firefox) where the native color
+                 picker stays visible when the modal is dismissed.
+
+                 - could not get modelbinder to work directly with spectrum; thus the colorChange
+                 function and re-running initSpectrum from resetDefaults.
+                 */
+                var $colorInputs = $('div[id="colorTab"] input[type="color"]');
+                var colorTabView = this;
+
+                var colorChange = function (spectrumTinyColor) {
+                    var newColor = spectrumTinyColor.toHexString();
+                    this.$colorInput.val(newColor);
+                };
+
+                $colorInputs.each(function () {
+                    var $color = $(this);
```

Comment:
You should be able to add two parameters to the callback function and this value I think will be set in the second parameter.

http://api.jquery.com/jQuery.each/